Feature/python virtual env hook

softyoungha opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

softyoungha committed:
update feedback

seyoon-lim commented:
@softyoungha 
우리 main 브랜치 업스트림이랑 싱크 맞춤요

seyoon-lim commented on airflow/hooks/python_virtualenv.py:
```diff
@@ -0,0 +1,350 @@
+        from flask_appbuilder.fieldwidgets import BS3TextAreaFieldWidget, BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import BooleanField, StringField
+        from wtforms.validators import InputRequired, Optional, any_of
```
```suggestion
        from wtforms.validators import InputRequired, Optional, AnyOf
```
이번에 개발하면서 느낀건데 훅쪽에서 Alias 쓰는 규칙을 잘 모르겠더라구요 ㅋㅋ 뭐는 클래스처럼쓰고 뭐는 함수처럼쓰고

softyoungha replied:
![image](https://github.com/user-attachments/assets/c47b5952-a6d6-454a-a6e3-63757fad07bd)
ㅋㅋㅋㅋ

seyoon-lim replied:
@softyoungha 
얘네 코드에도 보면 거의 `any_of`를 많이쓰더라구요 ㅋㅋ
머가 맞는 컨벤션인지 판단이 안섰습니다 ㅋㅋ

seyoon-lim commented on airflow/hooks/python_virtualenv.py:
```diff
@@ -0,0 +1,350 @@
+
+    def _requirements_list(self, exclude_cloudpickle: bool = False) -> list[str]:
+        """Prepare a list of requirements that need to be installed for the virtual environment."""
+        requirements = [str(dependency) for dependency in self.requirements]
```
```suggestion
        requirements = [dependency for dependency in self.requirements]
```
requirements가 Iterable[str] 타입으로 반환하니까 프로퍼티쪽에서 처리해주는게 좋을듯요

softyoungha replied:
이거 기존 PythonVirtualenvOperator 코드 그대로 복붙한건데, 뭔가 이유가 잇겟지요
그대로 가져갑니다
https://github.com/apache/airflow/blob/e30e1ee5401ab60e76eb34dd198385660a34faee/airflow/operators/python.py#L731-L744

seyoon-lim commented on airflow/hooks/python_virtualenv.py:
```diff
@@ -0,0 +1,350 @@
+
+    @contextmanager
+    def prepare_venv(self) -> Generator[Path, None, None]:
+        venv_cache_path = self.venv_cache_path
+        if venv_cache_path:
+            yield self._ensure_venv_cache_exists(Path(self.venv_cache_path))
```
```suggestion
        if self.venv_cache_path:
            yield self._ensure_venv_cache_exists(Path(self.venv_cache_path))
```
음? 차이점 있나여?

softyoungha replied:
아래 self를 빼야되는데 빼먹었네여
cached_property를 사용하는것에 대해서 한번 물어보긴하려구여
hook의 생성자에서 선언되는 속성들이 이후에 처음 초기화 후에 변경되는 점이 없다면 전체 cached_property로 하는 게 효율적이지 않나 싶은데, 별거 아닌 연산이여서 property로 두는것도 나쁘지 않을 것 같기도 하구여

seyoon-lim commented on airflow/utils/db.py:
```diff
@@ -559,12 +559,12 @@ def create_default_connections(session: Session = NEW_SESSION):
+            extra="""{
+    "requirements": "colorama==0.4.0",
+    "venv_cache_path": "/tmp/airflow"
+}""",
```
인덴트 쫌 불편하네여 😤

softyoungha replied:
![image](https://github.com/user-attachments/assets/98bcff1a-8768-46ec-961b-b6b0081d3e33)

바로 아래의 redshift connection 세팅과 맞춤요

seyoon-lim replied:
오 코드 잘못밨네여 ㅋㅋ
이거 json필드라 저렇게넣는군요
먼가 dict로 인덴트 맞게 넣은담에 json dump떠버리는게 좋을거같기도한데 위에 코드들도 저렇게되어있으면 요게 맞을듯요 ㅋㅋ

jscheffl committed:
Feature/add token authentication to internal api (#40899)

* Add Authentication via Token to Internal API

* Add Authentication via Token to Internal API

* Review feedback, use other signing method

* Fix pytest after change of token auth

* Review feedback, implement token with option b / own config value

* Add configuration pytest for additional secret

* Review Feedback, direct commit

Co-authored-by: Vincent <97131062+vincbeck@users.noreply.github.com>

---------

Co-authored-by: Vincent <97131062+vincbeck@users.noreply.github.com>

amoghrajesh committed:
Validating provider documentation urls before displaying in views (#40933)

Lee-W committed:
Extend get_datasets endpoint to include dataset aliases (#40830)

Lee-W committed:
Retrieve inlet dataset events through dataset aliases (#40809)

ephraimbuddy committed:
Fix log endpoint query broken in #40791 (#40950)

Turns out we can't combine the TI history and TI in a single query
and forcing it break things.

kacpermuda committed:
[AIP-62] Translate AIP-60 URI to OpenLineage (#40173)

* aip-62: implement translation mechanism from aip-60 to OpenLineage

Signed-off-by: Kacper Muda <mudakacper@gmail.com>

* aip-62: implement translation examples from aip-60 to OpenLineage

Signed-off-by: Kacper Muda <mudakacper@gmail.com>

---------

Signed-off-by: Kacper Muda <mudakacper@gmail.com>

dolfinus committed:
openlineage: Add AirflowRunFacet for dag runEvents (#40854)

JDarDagran committed:
openlineage: migrate OpenLineage provider to V2 facets. (#39530)

* Migrate OpenLineage proivder to V2 facets.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

* Migrate amazon to v2 facets.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

* Migrate OL-dependent providers to v2 facets.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

* Migrate Google provider to V2 facets.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

* Make migration backwards-compatible with previous OL provider versions.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

* Move V2 facets code imports to `common.compat` provider.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

* Add pre-commit hook for check on `common.compat` imports over OL v1.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

---------

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

potiuk committed:
Add more combination of SBOM files and fixes generation for new CDXGEN (#40959)

With the new command we generate 4 different sbom types with python/npm
only, with both python and npm where python variants are either basic
runtime dependencies of airflow or full dependencies including providers
and development dependencies.

It works with latest CDXGEN and thanks to use --parallel flag, we can
significantly speed up generation of those SBOMs.

Fixes: #40810

potiuk committed:
Improve ATTRIBUTE_REMOVED sentinel to use class and more context (#40920)

ATTRIBUTE_REMOVED was a singleton object but then it made it
difficult to find out which attribute has been removed. This PR
changes the approach to allow for multiple AttributeRemoved objects
as sentinel and compare class ratehr than object. The object contains
name of the removed attribute. Better diagnostics at the expense of
a bit more memory used.

kacpermuda committed:
openlineage: extend custom_run_facets to also be executed on complete and fail (#40953)

Signed-off-by: Kacper Muda <mudakacper@gmail.com>

softyoungha committed:
update

softyoungha committed:
update

softyoungha committed:
update test

softyoungha committed:
add db_test mark
